models/yolo/yolo_impl.hpp

```diff
@@ -0,0 +1,167 @@
+/**
+ * @file yolo_impl.hpp
+ * @author Kartik Dutt
+ *
+ * Implementation of YOLO models using mlpack.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MODELS_YOLO_IMPL_HPP
+#define MODELS_YOLO_IMPL_HPP
+
+#include "yolo.hpp"
+
+namespace mlpack {
+namespace ann {
+
+template<
+    typename OutputLayerType,
+    typename InitializationRuleType
+>
+YOLO<OutputLayerType, InitializationRuleType>::YOLO() :
+    inputChannel(0),
+    inputWidth(0),
+    inputHeight(0),
+    numClasses(0),
+    numBoxes(0),
+    featureWidth(0),
+    featureHeight(0),
+    weights("none"),
+    yoloVersion("none")
+{
+  // Nothing to do here.
+}
+
+template<
+    typename OutputLayerType,
+    typename InitializationRuleType
+>
+YOLO<OutputLayerType, InitializationRuleType>::YOLO(
+  const size_t inputChannel,
+  const size_t inputWidth,
+  const size_t inputHeight,
+  const std::string yoloVersion,
+  const size_t numClasses,
+  const size_t numBoxes,
+  const size_t featureWidth,
+  const size_t featureHeight,
+  const std::string& weights,
+  const bool includeTop) :
+  YOLO<OutputLayerType, InitializationRuleType>(
+    std::tuple<size_t, size_t, size_t>(
+      inputChannel,
+      inputWidth,
+      inputHeight),
+      yoloVersion,
+      numClasses,
+      numBoxes,
+      std::tuple<size_t, size_t>(featureWidth, featureHeight),
+      weights,
+      includeTop)
+{
+  // Nothing to do here.
+}
+
+template<
+     typename OutputLayerType,
+     typename InitializationRuleType
+>
```

Comment:
```suggestion
template<
    typename OutputLayerType,
    typename InitializationRuleType
>
```

Reply:
Sorry about the same style errors repeated in different PRs. I think I fixed them all. Thanks for pointing them out.